applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -1200,8 +1228,10 @@ static void update_key(const struct hid_keymap *map, int16_t value)
 	} else {
 		/* Update state and issue report generation event. */
 		if (key_value_set(&rd->items, map->usage_id, value)) {
-			rd->update_needed = true;
-			report_send(rd, false, true);
+			if (rd->linked_rs) {
+				rd->linked_rs->update_needed = true;
+			}
+			report_send(rd, rd->linked_rs, false, true);
```

Comment:
```
report_send(rd, NULL, false, true);
```
check changes in `report_send`
btw - I think explicitly setting update needed at this level is not needed - we do set always_send to true which will ensure the flag is set

Reply:
the same goes for lines 1300, 1316 (& friends)